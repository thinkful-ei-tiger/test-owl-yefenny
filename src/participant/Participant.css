.participant{
    display: flex;
}

.participant-img img{
    height: 50px;
    width: 50px;
    border-radius: 50%;
}

.participant-details{
    margin-left: 10px;
}

.participant-details h2 {
    font-size: 14px;
    margin-bottom: 0;
 
}

.participant-status{
    display: flex;
}

.circle {
    height: 11px;
    width: 11px;
    border-radius: 50%;
    margin-top: 7px;;
}

.bg-green {
    background-color: seagreen;
}

.bg-grey {
    background-color: silver;
}

.status{
    color: grey;
    font-size: 13.8px;
    margin-left: 8px;
    margin-top: 5px;
}