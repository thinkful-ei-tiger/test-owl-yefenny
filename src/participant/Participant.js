import React, { Component } from 'react';
import './Participant.css';

export default class Participant extends Component {
  render() {
    return (
      <div className='participant'>
        <div className='participant-img'>
          <img src={this.props.avatar} alt='avatar' />
        </div>
        <div className='participant-details'>
          <h2>{this.props.name}</h2>
          <div className='participant-status'>
            <span
              className={`circle ${
                this.props.inSession ? 'bg-green' : 'bg-grey'
              }`}
            ></span>
            <span className='status'>
              {this.props.inSession
                ? this.props.onStage
                  ? 'on stage'
                  : 'in session'
                : 'left session'}
            </span>
          </div>
        </div>
      </div>
    );
  }
}
