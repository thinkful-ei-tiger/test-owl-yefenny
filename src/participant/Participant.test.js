import React from 'react';
import renderer from 'react-test-renderer';
import Participant from './Participant';
import ReactDOM from 'react-dom';

describe('Participant component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Participant />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('Render UI without problem', () => {
    const tree = renderer.create(
      <Participant
        avatar='https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1'
        name='Koren Templeton'
        inSession={true}
        onStage={true}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
