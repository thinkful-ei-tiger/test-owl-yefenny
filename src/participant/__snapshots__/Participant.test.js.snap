// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participant component Render UI without problem 1`] = `
<div
  className="participant"
>
  <div
    className="participant-img"
  >
    <img
      alt="avatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
  </div>
  <div
    className="participant-details"
  >
    <h2>
      Koren Templeton
    </h2>
    <div
      className="participant-status"
    >
      <span
        className="circle bg-green"
      />
      <span
        className="status"
      >
        on stage
      </span>
    </div>
  </div>
</div>
`;
